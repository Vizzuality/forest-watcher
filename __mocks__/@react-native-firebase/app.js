import RNFBAnalytics from './analytics';

export default {
  analytics: () => RNFBAnalytics,
  getFirebaseRoot: () => {}
};
