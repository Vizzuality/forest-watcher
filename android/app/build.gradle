apply plugin: "com.android.application"
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"

import com.android.build.OutputFile

apply from: "../../node_modules/react-native/react.gradle"

def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 23
    buildToolsVersion '25.0.0'

    defaultConfig {
        applicationId "com.forestwatcher"
        minSdkVersion 17
        targetSdkVersion 22
        versionCode 3
        versionName "1.2"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
        resValue "string", "build_config_package", "com.forestwatcher"
    }
    signingConfigs {
        debug {
            storeFile file(project.env.get("RELEASE_STORE_FILE"))
            storePassword project.env.get("RELEASE_STORE_PASSWORD")
            keyAlias project.env.get("RELEASE_KEY_ALIAS")
            keyPassword project.env.get("RELEASE_KEY_PASSWORD")
        }
        release {
            storeFile file(project.env.get("RELEASE_STORE_FILE"))
            storePassword project.env.get("RELEASE_STORE_PASSWORD")
            keyAlias project.env.get("RELEASE_KEY_ALIAS")
            keyPassword project.env.get("RELEASE_KEY_PASSWORD")
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    compile(project(':react-native-google-signin')){
      exclude group: "com.google.android.gms"
    }
    compile project(':react-native-google-analytics-bridge')
    compile project(':react-native-sensor-manager')
    compile project(':react-native-camera')
    compile project(':react-native-fetch-blob')
    compile project(':react-native-i18n')
    compile project(':react-native-config')
    compile project(':RNLocation')
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
    compile(project(':react-native-maps')){
      exclude group: 'com.google.android.gms', module: 'play-services-base'
      exclude group: 'com.google.android.gms', module: 'play-services-maps'
    }
    compile 'com.google.android.gms:play-services-base:10.2.1'
    compile 'com.google.android.gms:play-services-maps:10.2.1'
    compile 'com.google.android.gms:play-services-auth:10.2.1'
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}

apply plugin: 'com.google.gms.google-services'
