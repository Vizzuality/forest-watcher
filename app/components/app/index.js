import React from 'react';
import {
  View,
  StatusBar
} from 'react-native';

import { StackNavigator } from 'react-navigation';
import { Routes, RoutesConfig } from 'routes';

import Login from 'containers/login';
import styles from './styles';

function App() {
  StatusBar.setBarStyle('default', true);

  const AppNavigator = StackNavigator(Routes, RoutesConfig);
  return (
    <View style={styles.mainContainer}>
      <AppNavigator />
      <Login />
    </View>
  );
}

App.propTypes = {};

export default App;
