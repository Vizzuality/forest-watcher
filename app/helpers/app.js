// @flow
import { PermissionsAndroid, Platform } from 'react-native';
import RNLocation from 'react-native-location';

import type { State } from 'types/store.types';

import { STATUS } from 'config/constants/index';

import { version } from 'package.json';

// eslint-disable-next-line import/default
import codePush from 'react-native-code-push';

export function isUnsafeLogout(state: State) {
  const { list } = state.reports;
  const hasReportsToUpload = type => type === STATUS.complete || type === STATUS.draft;
  return Object.values(list)
    .map(report => report.status)
    .some(hasReportsToUpload);
}

export function shouldBeConnected(state: State) {
  const { offline, app } = state;
  return offline.online && !app.offlineMode;
}

export function getVersionName() {
  return codePush.getUpdateMetadata().then(update => {
    if (update) {
      return `v${version}-codepush_${update.label}`;
    }
    return `v${version}`;
  });
}
