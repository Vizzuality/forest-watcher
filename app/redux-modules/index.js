export { reducer as form } from 'redux-form';
export { default as alerts } from './alerts';
export { default as user } from './user';
export { default as countries } from './countries';
export { default as setup } from './setup';
export { default as areas } from './areas';
export { default as reports } from './reports';
