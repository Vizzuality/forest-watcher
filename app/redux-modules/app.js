// @flow
import type { Dispatch, GetState, Action } from 'types/store.types';
import type { AppState, AppAction, CoordinatesValue } from 'types/app.types';

// $FlowFixMe
import { version } from 'package.json'; // eslint-disable-line
import { COORDINATES_FORMATS, ACTIONS_SAVED_TO_REPORT } from 'config/constants/index';
import { syncAreas } from 'redux-modules/areas';
import { getLanguage } from 'helpers/language';
import { syncCountries } from 'redux-modules/countries';
import { syncUser, LOGOUT_REQUEST } from 'redux-modules/user';
import { syncLayers } from 'redux-modules/layers';
import { syncReports } from 'redux-modules/reports';
import { PERSIST_REHYDRATE } from '@redux-offline/redux-offline/lib/constants';
import takeRight from 'lodash/takeRight';

// Actions
const SET_OFFLINE_MODE = 'app/SET_OFFLINE_MODE';
export const SET_APP_SYNCED = 'app/SET_APP_SYNCED';
export const RETRY_SYNC = 'app/RETRY_SYNC';
const SET_COORDINATES_FORMAT = 'app/SET_COORDINATES_FORMAT';
const SET_PRISTINE_CACHE_TOOLTIP = 'app/SET_PRISTINE_CACHE_TOOLTIP';
const SET_WELCOME_SEEN = 'app/SET_WELCOME_SEEN';
export const SAVE_LAST_ACTIONS = 'app/SAVE_LAST_ACTIONS';
export const SHOW_OFFLINE_MODE_IS_ON = 'app/SHOW_OFFLINE_MODE_IS_ON';
export const SHOW_CONNECTION_REQUIRED = 'app/SHOW_CONNECTION_REQUIRED';
export const UPDATE_APP = 'app/UPDATE_APP';
export const EXPORT_REPORTS_SUCCESSFUL = 'app/EXPORT_REPORTS_SUCCESSFUL';

// Reducer
const initialState = {
  version,
  actions: [],
  synced: false,
  language: getLanguage(),
  offlineMode: false,
  pristineCacheTooltip: true,
  coordinatesFormat: COORDINATES_FORMATS.decimal.value,
  hasSeenWelcomeScreen: false
};

export default function reducer(state: AppState = initialState, action: AppAction) {
  switch (action.type) {
    case PERSIST_REHYDRATE: {
      // $FlowFixMe
      const { app } = action.payload;
      const language = getLanguage();
      const isUpdate = app?.version != undefined && app?.version != null && app?.version != version;
      return { ...state, ...app, language, version, isUpdate };
    }
    case SET_OFFLINE_MODE:
      return { ...state, offlineMode: action.payload };
    case SET_APP_SYNCED:
      return { ...state, synced: action.payload };
    case SET_COORDINATES_FORMAT:
      return { ...state, coordinatesFormat: action.payload };
    case SET_PRISTINE_CACHE_TOOLTIP:
      return { ...state, pristineCacheTooltip: action.payload };
    case SET_WELCOME_SEEN:
      return { ...state, hasSeenWelcomeScreen: action.payload };
    case SAVE_LAST_ACTIONS: {
      // Save the last actions to send to the report
      const actions = [...takeRight(state.actions, ACTIONS_SAVED_TO_REPORT), action.payload];
      return { ...state, actions };
    }
    case LOGOUT_REQUEST:
      return {
        ...initialState,
        language: state.language
      };
    default:
      return state;
  }
}

export function setOfflineMode(offlineMode: boolean): AppAction {
  return {
    type: SET_OFFLINE_MODE,
    payload: offlineMode
  };
}

export function setAppSynced(open: boolean): AppAction {
  return {
    type: SET_APP_SYNCED,
    payload: open
  };
}

export function syncApp() {
  return (dispatch: Dispatch, state: GetState) => {
    const { user } = state();
    dispatch(syncUser());
    if (user.loggedIn) {
      dispatch(syncCountries());
      dispatch(syncReports());
      dispatch(syncAreas());
      dispatch(syncLayers());
    }
  };
}

export function updateApp() {
  return {
    type: UPDATE_APP
  };
}

export function retrySync() {
  return (dispatch: Dispatch) => {
    dispatch({ type: RETRY_SYNC });
    dispatch(syncApp());
  };
}

export function saveLastActions(payload: Action) {
  return {
    type: SAVE_LAST_ACTIONS,
    payload
  };
}

export function setCoordinatesFormat(format: CoordinatesValue): AppAction {
  return {
    type: SET_COORDINATES_FORMAT,
    payload: format
  };
}

export function setWelcomeScreenSeen(seen: boolean): AppAction {
  return {
    type: SET_WELCOME_SEEN,
    payload: seen
  };
}

export function setPristineCacheTooltip(pristine: boolean): AppAction {
  return {
    type: SET_PRISTINE_CACHE_TOOLTIP,
    payload: pristine
  };
}

export function showNotConnectedNotification() {
  return (dispatch: Dispatch, getState: GetState) => {
    const { offlineMode } = getState().app;
    if (offlineMode) {
      return dispatch({ type: SHOW_OFFLINE_MODE_IS_ON });
    }
    return dispatch({ type: SHOW_CONNECTION_REQUIRED });
  };
}

export function showExportReportsSuccessfulNotification() {
  return (dispatch: Dispatch, getState: GetState) => {
    return dispatch({ type: EXPORT_REPORTS_SUCCESSFUL });
  };
}
