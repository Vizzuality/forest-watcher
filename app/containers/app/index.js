import { connect } from 'react-redux';
import { navigatePush, navigatePop } from 'redux-modules/navigation';
import { setLoginModal, setLoginStatus } from 'redux-modules/user';
import App from 'components/app';

function mapStateToProps(state) {
  return {
    navigationRoute: state.navigation,
    loggedIn: state.user.loggedIn
  };
}

function mapDispatchToProps(dispatch) {
  return {
    navigate: (action) => {
      dispatch(navigatePush(action));
    },
    navigateBack: () => {
      dispatch(navigatePop());
    },
    setLoginModal: (action) => {
      dispatch(setLoginModal(action));
    },
    setLoginStatus: (action) => {
      dispatch(setLoginStatus(action));
    }
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);
