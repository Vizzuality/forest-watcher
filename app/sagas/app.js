// @flow
import { takeEvery, put } from 'redux-saga/effects';
import { saveLastActions, SAVE_LAST_ACTIONS } from 'redux-modules/app';

declare var __DEV__: boolean;

export function* logLastActions(): Generator<*, *, *> {
  yield takeEvery('*', function* logger(action) {
    const log = action.type !== SAVE_LAST_ACTIONS && !action.type.startsWith('user/');
    if (log && !__DEV__) {
      yield put(saveLastActions(action));
    }
  });
}
