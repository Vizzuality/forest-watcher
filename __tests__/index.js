import 'react-native';
import React from 'react';
import { default as reducer, setOfflineMode, syncApp } from '../app/redux-modules/app';

describe('Unit Tests', () => {

  it('test runs', () => {
    expect(true);
  });

  it('snapshot runs', () => {
    expect({'hello': true, number: 232}).toMatchSnapshot();
  });

  it('snapshot redux action test', () => {
    expect(setOfflineMode(true)).toMatchSnapshot();
    expect(setOfflineMode(false)).toMatchSnapshot();
  });

  it('snapshot redux reducer test', () => {
    expect(reducer(undefined, setOfflineMode(true))).toMatchSnapshot();
  });

  it('snapshot sync app function test', () => {
    expect(syncApp()).toMatchSnapshot();
  });

});

// MOCKS

jest.mock('react-native-config', () => {return {API_URL: '', MAPBOX_TOKEN: ''}});
jest.mock('react-native-cookies', () => {clearAll: jest.fn()});
jest.mock('react-native-fetch-blob', () => {config: jest.fn()});
jest.mock('react-native-zip-archive', () => {unzip: jest.fn()});
jest.mock('Dimensions');
