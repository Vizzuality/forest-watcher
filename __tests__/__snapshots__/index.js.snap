// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests snapshot redux action test 1`] = `
Object {
  "payload": true,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Unit Tests snapshot redux action test 2`] = `
Object {
  "payload": false,
  "type": "app/SET_OFFLINE_MODE",
}
`;

exports[`Unit Tests snapshot redux reducer test 1`] = `
Object {
  "actions": Array [],
  "coordinatesFormat": "decimal",
  "language": null,
  "offlineMode": true,
  "pristineCacheTooltip": true,
  "synced": false,
  "version": "1.6.3",
}
`;

exports[`Unit Tests snapshot runs 1`] = `
Object {
  "hello": true,
  "number": 232,
}
`;

exports[`Unit Tests snapshot sync app function test 1`] = `[Function]`;
